cylc/flow/job_file.py

```diff
@@ -56,10 +55,9 @@ def write(self, local_job_file_path, job_conf, check_syntax=True):
         # Access to cylc must be configured before user environment so
         # that cylc commands can be used in defining user environment
         # variables: NEXT_CYCLE=$( cylc cycle-point --offset-hours=6 )
-
-        tmp_name = local_job_file_path + '.tmp'
-        run_d = get_remote_suite_run_dir(
-            job_conf['host'], job_conf['owner'], job_conf['suite_name'])
+        platform = job_conf['platform']
+        tmp_name = os.path.expandvars(local_job_file_path + '.tmp')
```

Comment:
They've been added because Matt had some very clever logic in `get_host_item` to determine whether to expand variables on a case by-case-basis, which has been removed. The removal of this logic has been the single biggest cause of pain in this PR.

Reply:
FYI this is the decentralisation of `$HOME` expansion logic discussed on Riot.

We can niceify as follow on, I've created an issue here: https://github.com/cylc/cylc-flow/issues/3767